src/Stats/index.jsx

```diff
@@ -51,12 +53,16 @@ export default function Stats() {
       <ScrollView style={styles.deviceBody}>
         <View style={{ marginBottom: 20 }}>
           <View style={{ flexDirection: 'row', justifyContent: 'space-between' }}>
-            <FilterBtn />
+            <FilterBtn
+              transFilter={transFilter}
+              setTransFilter={setTransFilter}
+              groupTransactions={filterListView(transactions)}
+            />
           </View>
           {(transactions.length !== 0)
             ? (
               <TransList
-                transactions={filterListView(utils.groupTransactionsByDate(transactions))}
+                transactions={filterListView(utils.filterType(transactions, transFilter))}
```

Comment:
Let's rename the `filterListView` function to `filterTransactions` this method will apply all filters, like type and date. Before rendering you can call this method, which returns all the transactions what we want to display in a flat array. So with this array you can calculate the sum of expenses/incomes which can be passed down to the `FilterBtn` component (So you can remove the calculation from that component) Finally after you calculated the sums you can group the transactions by date and pass it the to `TransList` component.
```js
function filterTransactions(transactions) {
  const transactionsWithSpecificType = filterTransactionsByType(transactions);
  return filterTransactionsByDate(transactionsWithSpecifictype);
}

function calculateSumByType(transactions, type) {
  ...
}

const filteredTransactions = filterTransactions(transactions);
const totalfExpense = calculate...
const totalIncome = calculate...
const processedTransactions = groupTransactionsByDate(filteredTransactions);

....
```

Reply:
fixed